Test/Helper/StoreTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NS8\Protect\Test;
+
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Store\Model\ResourceModel\Store\Collection;
+use Magento\Framework\App\Helper\Context;
+use Magento\Store\Model\StoreFactory;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Store\Model\Store;
+use NS8\Protect\Helper\Store as StoreHelper;
+use PHPUnit\Framework\TestCase;
+use Zend\Uri\Uri;
+
+/**
+ * Tests for Helper\Store.php
+ */
+class StoreTest extends TestCase
+{
+    /** @var Config */
+    private $config;
+
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $objectManager = new ObjectManager($this);
+        /** @var Context */
+        $context = $this->createMock(Context::class);
+
+        $store1 = [
+            'name'  => 'store1',
+            'code'  => 's1',
+            'id'    => 1,
+            'url'   => 'store1.com',
+            'active' => true
+        ];
+
+        $store2 = [
+            'name'  => 'store2',
+            'code'  => 's2',
+            'id'    => 2,
+            'url'   => 'store2.com',
+            'active' => true
+        ];
+
+        $store1Mock = $this->getMockBuilder(Store::class)
```

Comment:
this whole block might be best pushed out to a `TestHelper` or something, I imagine we'll be mocking a lot of stores and it'll be useful here regardless